witchcraft-server/src/logging/logger/mod.rs

```diff
@@ -35,8 +36,11 @@ pub mod rolling_file;
 pub mod stdout;
 
 pub type Appender<T> = AsyncAppender<T>;
-pub type SyncAppender<T> =
-    MetricsAppender<JsonAppender<Pin<Box<dyn Sink<Bytes, Error = io::Error> + Sync + Send>>>, T>;
+
+pub struct Payload<T> {
+    pub value: T,
+    pub cb: Option<oneshot::Sender<bool>>,
```

Comment:
What's the reason we have the Sender take a `bool` instead of something like a `Result<(), dyn Error>`?

Reply:
Errors aren't cloneable and a single error will often need to be reported to many payloads. Even if we could clone the errors, we wouldn't want to log the same error 100 times or whatever for each failed payload.